src/main/java/io/github/bobocodebreskul/context/scan/RecursiveClassPathAnnotatedBeanScanner.java

```diff
@@ -37,6 +37,7 @@ public RecursiveClassPathAnnotatedBeanScanner(ScanUtils scanUtils,
 
   @Override
   public void scan(String... scanPackages) {
+    scanUtils.validatePackagesToScan(scanPackages);
```

Comment:
I recommend to validate packages directly in the `ScanUtilImpl` class all search methods.